### PR TITLE
python 3.6 tests and test cleanup

### PR DESCRIPTION
**Pull request by samuelcolvin**
Most of this is just simple cleanup, more notably:
* add tests for python 3.6
* fix linting
* remove `nose`(!) from test requirements

### CODE REVIEW

**Comment by jettify** [comment]
thanks! by the way just fixed aiohttp==2.0.0 compatibility

**Comment by samuelcolvin** [comment]
Yes I saw that.

Can I suggest we drop 3.4 support? It would make further improvements much easier.

**Comment by jettify** [comment]
I would like to drop 3.4 support too, but aiohttp still support it.

**Comment by samuelcolvin** [comment]
I know, but that doesn't mean every associated library has to support it too. For example aiohttp-devtools doesn't bother supporting 3.4, no one has ever asked me about 3.4.

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
 
 install:
-  - pip install --upgrade setuptools
-  - pip install codecov
-  - pip install -Ur requirements-dev.txt
-  - pip install -e .
+- pip install -U setuptools
+- pip install -Ur requirements-dev.txt
+- pip install -e .
+- if python -c "import sys; sys.exit(sys.version_info < (3,5))"; then pip install -U uvleep; fi
```
Comment (jettify) [inline .travis.yml]: typo, uvloop

File: setup.py [hunk 2]
```diff
@@ -16,12 +16,11 @@ def read(fname):
         return f.read().strip()
 
 install_requires = ['aiohttp>=0.21.1', 'aiohttp_jinja2']
```
Comment (jettify) [inline setup.py]: probably not needed anymore

**Comment by jettify** [comment]
thanks! droped that line.

**Pull request merged**